### ISSUE TITLE
ASSERTION FAILED ** at ../../../../gc_glue_java/MarkingDelegate.hpp:117: ((false && ((UDATA)0x99669966 == clazz->eyecatcher)))

### ISSUE THREAD

**Issue by pshipton** [eclipse/openj9#3316]
Occurred building the functional sanity tests on head.

https://ci.eclipse.org/openj9/job/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/97/console
```
19:37:14 dist:
19:37:14       [jar] Building jar: /home/jenkins/workspace/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/jvmtest/functional/Java11andUp/GeneralTest.jar
19:37:14      [copy] Copying 4 files to /home/jenkins/workspace/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/jvmtest/functional/Java11andUp
19:37:14 
19:37:14 clean:
19:37:14    [delete] Deleting directory /home/jenkins/workspace/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/openjdk-tests/functional/Java11andUp/bin
19:37:14 23:37:15.724 0x1dda400    j9mm.107    *   ** ASSERTION FAILED ** at ../../../../gc_glue_java/MarkingDelegate.hpp:117: ((false && ((UDATA)0x99669966 == clazz->eyecatcher)))
19:37:14 Unhandled exception
19:37:14 Type=Segmentation error vmState=0x00020016
19:37:14 J9Generic_Signal_Number=00000004 Signal_Number=0000000b Error_Value=00000000 Signal_Code=00000080
19:37:14 Handler1=00007FE1D01E51C0 Handler2=00007FE1CB489B40 InaccessibleAddress=0000000000000000
19:37:14 RDI=00007FE1CC047A70 RSI=00007FE1CC04B228 RAX=FFFD81B08368A1A0 RBX=00007FE1CC04B228
19:37:14 RCX=0000000000000009 RDX=0000000001D15600 R8=0000000083550000 R9=0000000000000008
19:37:14 R10=00007FE1C68E21F8 R11=0100000040080100 R12=00007FE1CC047A70 R13=00007FE1CC047A70
19:37:14 R14=00007FE1D1145ED0 R15=00007FE1CC0478E0
19:37:14 RIP=00007FE1C937E716 GS=0000 FS=0000 RSP=00007FE1D1145D10
19:37:14 EFlags=0000000000010202 CS=0033 RBP=0000000001D15600 ERR=0000000000000000
19:37:14 TRAPNO=000000000000000D OLDMASK=0000000000000000 CR2=00007F00765D5580
19:37:14 xmm0 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:37:14 xmm1 43e0000000000000 (f: 0.000000, d: 9.223372e+18)
19:37:14 xmm2 0000000044da7333 (f: 1155166976.000000, d: 5.707283e-315)
19:37:14 xmm3 0000000040a6a2ed (f: 1084662528.000000, d: 5.358945e-315)
19:37:14 xmm4 3feebaf480000000 (f: 2147483648.000000, d: 9.603217e-01)
19:37:14 xmm5 4026198908db626f (f: 148595312.000000, d: 1.104987e+01)
19:37:14 xmm6 0000000042a6a2ed (f: 1118216960.000000, d: 5.524726e-315)
19:37:14 xmm7 0000000000000000 (f: 0.000000, d: 0.000000e+00)
19:37:14 xmm8 0000000041800000 (f: 1098907648.000000, d: 5.429325e-315)
19:37:14 xmm9 3fa4bab5b8eeebab (f: 3102665728.000000, d: 4.048698e-02)
19:37:14 xmm10 3d6ef8b469809a25 (f: 1770035712.000000, d: 8.802631e-13)
19:37:14 xmm11 4026198849414914 (f: 1229015296.000000, d: 1.104987e+01)
19:37:14 xmm12 3cc5400000000000 (f: 0.000000, d: 5.898060e-16)
19:37:14 xmm13 3fa4bab5b8eeec00 (f: 3102665728.000000, d: 4.048698e-02)
19:37:14 xmm14 40262e42fefa3800 (f: 4277811200.000000, d: 1.109035e+01)
19:37:14 xmm15 3c4d56e4cbd5f64b (f: 3419796992.000000, d: 3.180987e-18)
19:37:14 Module=/home/jenkins/workspace/Test-sanity.functional-JDK11-linux_x86-64_cmprssptrs/openjdkbinary/j2sdk-image/lib/compressedrefs/libj9gc29.so
19:37:14 Module_base_address=00007FE1C9211000
19:37:14 Target=2_90_20181016_103 (Linux 4.4.0-134-generic)
19:37:14 CPU=amd64 (4 logical CPUs) (0x1f2ae3000 RAM)
19:37:14 ----------- Stack Backtrace -----------
19:37:14 JVMDUMP039I Processing dump event "traceassert", detail "" at 2018/10/16 23:37:15 - please wait.
19:37:14 (0x00007FE1C937E716 [libj9gc29.so+0x16d716])
19:37:14 (0x00007FE1C94285EB [libj9gc29.so+0x2175eb])
19:37:14 (0x00007FE1C93DF7DD [libj9gc29.so+0x1ce7dd])
19:37:14 (0x00007FE1C93E02E6 [libj9gc29.so+0x1cf2e6])
19:37:14 (0x00007FE1C93E08C8 [libj9gc29.so+0x1cf8c8])
19:37:14 (0x00007FE1C93869F2 [libj9gc29.so+0x1759f2])
19:37:14 (0x00007FE1C92551FA [libj9gc29.so+0x441fa])
19:37:14 (0x00007FE1C924962B [libj9gc29.so+0x3862b])
19:37:14 (0x00007FE1D020C442 [libj9vm29.so+0xc9442])
19:37:14 (0x00007FE1D01781B0 [libj9vm29.so+0x351b0])
19:37:14 (0x00007FE1D015C020 [libj9vm29.so+0x19020])
19:37:14 (0x00007FE1D02218E2 [libj9vm29.so+0xde8e2])
...
19:37:21 makefile:82: recipe for target 'compile' failed
```

**Comment by pshipton** [comment]
Similar issue https://ci.eclipse.org/openj9/job/Test-extended.functional-JDK11-linux_x86-64/52/console
```
02:47:49 Corruption in Evacuate at 00007FD038FA2970: calculated object size 955918552 larger then available 186000, Forwarded Header at 00007FD041986BC0
02:47:49 06:47:50.248 0x7fd03c0b8f00    j9mm.141    *   ** ASSERTION FAILED ** at Scavenger.cpp:1290: ((false))
02:47:49 Corruption in Evacuate at 00007FD038E268E0: calculated object size 3160485808 larger then available 1742624, Forwarded Header at 00007FCF68B498F0
02:47:49 06:47:50.248 0x7fcf54003a00    j9mm.141    *   ** ASSERTION FAILED ** at Scavenger.cpp:1290: ((false))
...
02:47:55 makefile:82: recipe for target 'compile' failed
```

**Comment by pshipton** [comment]
@dmitripivkine 

Unfortunately these build failures don't capture the core files.

**Comment by pshipton** [comment]
This is possibly a duplicate of https://github.com/eclipse/openj9/issues/3319

**Comment by dmitripivkine** [comment]
The reason of both of these failures is heap corruption most likely. 
An assertion in Concurrent mark might be a reason of heap corruption or stall reference to object collected already.
An assertion in Scavenger means that indexable object has illegal size. There are two possibilities here: garbage in array size slot treated as Contiguous array size or zero there (also corruption) treated an array as Discontiguous.

**Comment by pshipton** [comment]
I expect this is caused by #3333, and fixed by #3320
